specs/_features/maxeb_increase/capella.py

```diff
@@ -1627,14 +1670,42 @@ def process_pending_balance_deposits(state: BeaconState) -> None:
     state.pending_balance_deposits = state.pending_balance_deposits[next_pending_deposit_index:]
 
 
+def get_active_balance(state: BeaconState, validator: Validator) -> Gwei:
+    active_balance_ceil = MIN_ACTIVATION_BALANCE if has_eth1_withdrawal_credential(validator) else MAX_EFFECTIVE_BALANCE
+    return min(state.balances[validator.index], active_balance_ceil)
+
+def apply_pending_consolidation(state: BeaconState, pending_consolidation: PendingConsolidation) -> None:
+    source_validator = state.validators[pending_consolidation.source_index]
+    target_validator = state.validators[pending_consolidation.target_index]
+    # Move consolidation balance to target. Excess balance will be withdrawn.
+    active_balance = get_active_balance(state, source_validator)
+    state.balances[source_validator.index] -= active_balance
+    state.balances[target_validator.index] += active_balance
+
+
+def process_pending_consolidations(state: BeaconState) -> None:
+    next_pending_consolidation = 0
+    for pending_consolidation in state.pending_consolidations:
+        source_validator = state.validators[pending_consolidation.source_index]
+        if source_validator.withdrawable_epoch > get_current_epoch(state):
```

Comment:
If a validator gets slashed it will make processing of pending consolidations get stuck. For simplicity, we can simply drop consolidations of slashed validators. Alternatively, we can move consolidations of slashed validators to the end of the queue but it makes the queue management a bit more complicated, because likely such movement will be happening many times for the same source if it gets slashed

Reply:
Also consolidation alleviates slashing punishment for a bit. This is because the rest of the balance after penalties will be active again bypassing depositing and the activation queue. I think it is another slight argument in favour of just dropping slashed source’s from the queue

Reply:
Agreed! 

```suggestion
        if source_validator.slashed:
            next_pending_consolidation += 1
            continue
        if source_validator.withdrawable_epoch > get_current_epoch(state):
```